src/stores/index.js

```diff
@@ -245,8 +252,37 @@ const Store = Reflux.createStore({
     this.trigger(this.state);
   },
 
+  onCancelConnectionAttemptClicked() {
+    this._cancelCurrentConnectionAttempt();
+  },
+
   /**
-   * Resests URL validation.
+   * Cancels the current connection attempt if there is one
+   * in progress, and ends the connection.
+   */
+  async _cancelCurrentConnectionAttempt() {
+    if (this.state.isConnected || !this.state.isConnecting) {
+      return;
+    }
+
+    if (this.connectingConnectionAttempt) {
+      try {
+        this.connectingConnectionAttempt.cancelConnectionAttempt();
+      } catch (err) {
+        // When the disconnect fails, we free up the ui and we can
+        // silently wait for the timeout if it's still attempting to connect.
+      }
+
+      this.connectingConnectionAttempt = null;
+    }
+
+    this.setState({
+      isConnecting: false
+    });
+  },
+
+  /**
+   * Resets URL validation.
    */
   onConnectionFormChanged() {
     const currentConnection = this.state.currentConnection;
```

Comment:
sorry for being so picky .. i'm aware that most of those concepts were here already, feel free to ignore this if is too silly, but what about renaming `currentConnection`, `prop.currentConnection` and `state.currentConnection` etc .. into something that is more close to reality. `currentConnection` makes me think about the actual connection object from the driver (like the mongoclient) or similar .., would not really think of that being just an object with the connection attributes. What about `.connectionModel` instead?

Reply:
Going to rename currentConnection in a different pr since it's everywhere. I'll open it after this is merged if that's cool.